spec/controllers/admin/orders_controller_spec.rb

```diff
@@ -232,5 +232,65 @@ def send_request(order:)
         expect(controller.send(:authorized_action?, o, :destroy)).to be false
       end
     end
+
+    context "when the user is an admin" do
+      controller(Admin::OrdersController) do
+        def pundit_user
+          Customer.find_by(name: "Current User")
+        end
+      end
+
+      let!(:user) { create(:customer, name: "Current User") }
+
+      describe "GET new" do
+        it "allows me to new my records" do
+          expect { get :new }.not_to raise_error
+        end
+      end
+
+      describe "POST create" do
+        def send_request
+          post(
+            :create,
+            params: {
+              order: attributes_for(:order, customer: nil),
+            },
+          )
+        end
+
+        it "allows me to create my records" do
+          send_request
+          expect(response).to redirect_to([:admin, (order = Order.last)])
+          expect(order.customer).to eq(user)
+        end
+      end
+    end
+  end
+
+  context "when the user is not an admin" do
+    let!(:user) { create(:customer, name: "Target User") }
+
+    describe "GET new" do
+      it "allows me to new records" do
+        expect { get :new }.not_to raise_error
+      end
+    end
+
+    describe "POST create" do
+      def send_request
+        post(
+          :create,
+          params: {
+            order: attributes_for(:order, customer: nil, customer_id: user.id),
```

Comment:
Why not do this with the customer object? I feel it would be more natural. I had to read this a couple of times and check the factory to see what was going on:

```suggestion
            order: attributes_for(:order, customer: user),
```

Reply:
Since this is a request parameter, I was unable to format it as `customer: user`.
However, this spec had many mistakes overall, so I have made corrections.